### ISSUE TITLE
Warning using algorithm 2

### ISSUE THREAD

**Issue by jbruxaux** [Zilong-Li/PCAone#6]
Hi!

I am currently using PCAone on a fairly big dataset (195 samples, 4,2 million SNPs), and I get this warning:
`WARNING: PCAone converged but the window size did not expand the whole data.`
What would you recommend? Using a different algorithm?
I am using the last version of PCAone (v0.3.4) on a linux cluster.

Thanks for your help!

**Comment by Zilong-Li** [comment]
Hi!

Thanks for trying out PCAone. 

Mathmatically, without the warning, we would get `A ≈ USV` where `U` matrix is the PCs in `pcaone.eigvecs` file, S in `pcaone.eigvals` and `V` matrix is in the SNP loadings file `pcaone.loadings`. This warning literately means that PCAone converged in few epochs before the algorithm expands to the whole data, which suggests only U is correct while S and V are not meaningful.  

If you care about S and V matrix other than the U , you can force PCAone algorithm2 to run exact 7 epochs where it starts expanding to the whole data `A` with default settings. 

```@bash
PCAone --bfile plink -m 2 -n 20 -k 10  --svd 2 --tol-rsvd 0 --maxp 7
```

In fact, you data is quite small :) 


Best,
Zilong

**Comment by jbruxaux** [comment]
Hi Zilong,

I just re-ran the analysis as you suggested, and it worked without any issue.
Thanks for your help!

**Comment by Zilong-Li** [comment]
Sorry for the confusion. I'll update PCAone in the next release to make algorithm2 running at least 7 epochs although it converges in less epochs.

**Comment by Zilong-Li** [comment]
this is fixed in the latest release v0.3.5
